libraries/eosiolib/eosiolib.cpp

```diff
@@ -58,4 +61,70 @@ namespace eosio {
      return active_prods;
    }
 
+   // powers.hpp
+   template const std::array<uint64_t, 20>  powers_of_base<10, uint64_t>;
+
+   /**
+    *  Writes a number as a string to the provided char buffer
+    *
+    *  @brief Writes number x 10^(-num_decimal_places) (optionally negative) as a string to the provided char buffer
+    *  @pre The range [begin, end) must be a valid range of memory to write to.
+    *  @param begin - The start of the char buffer
+    *  @param end - Just past the end of the char buffer
+    *  @param dry_run - If true, do not actually write anything into the range.
+    *  @param number - The number to print before shifting the decimal point to the left by num_decimal_places.
+    *  @param num_decimal_places - The number of decimal places to shift the decimal point.
+    *  @param negative - Whether to print a minus sign in the front.
+    *  @return char* - Just past the end of the last character that would be written assuming dry_run == false and end was large enough to provide sufficient space. (Meaning only applies if returned pointer >= begin.)
+    *  @post If the output string fits within the range [begin, end), the range [begin, returned pointer) contains the string representation of the number. Nothing is written if dry_run == true or returned pointer > end (insufficient space) or if returned pointer < begin (overflow in calculating desired end).
+    */
+   char* write_decimal( char* begin, char* end, bool dry_run, uint64_t number, uint8_t num_decimal_places, bool negative ) {
+      constexpr static uint8_t log10_max_uint64 = powers_of_base<10, uint64_t>.size() - 1; // 19
```

Comment:
I don't see this being used anywhere, did I miss something?

Reply:
Yes, it is used in `asset.hpp`.